In [1]:
import os
import sys
import matplotlib.pyplot as plt
import numpy as np
import re

In [2]:
logfile = '/mnt/scratch/baburish/doublepulse/gnn/Analysis/logs/22644slurm_GAT.log'

with open(logfile, 'r') as f:
    lines = f.readlines()

In [3]:
#Epoch   3 | Train: 0.7505 | Val: 0.7251 | AUC: 0.7451 | Best: 0.7451
total_epochs = 110
epochs = []
train_loss = []
val_loss = []
auc = []
pattern = r"Epoch\s+(\d+)\s+\|\s+Train:\s+([\d\.]+)\s+\|\s+Val:\s+([\d\.]+)\s+\|\s+AUC:\s+([\d\.]+)\s+\|\s+Best:\s+([\d\.]+)"
for line in lines:
    match = re.search(pattern, line)
    if match:
        epochs.append(int(match.group(1)))
        train_loss.append(float(match.group(2)))
        val_loss.append(float(match.group(3)))
        auc.append(float(match.group(4)))

In [ ]:
fig, ax = plt.subplots(figsize=(10, 6))

ax.plot(epochs, train_loss, label='Train Loss', color='blue')
ax.plot(epochs, val_loss, label='Val Loss', color='red')
ax.set_xlabel('Epochs')
ax.set_ylabel('Loss')
ax2 = ax.twinx()
ax2.plot(epochs, auc, label='AUC', color='orange')
ax2.set_ylabel('AUC')
ax.set_title('Training and Validation Loss with AUC over Epochs')
# Combine legends from both axes
lines, labels = ax.get_legend_handles_labels()
lines2, labels2 = ax2.get_legend_handles_labels()
ax2.legend(lines + lines2, labels + labels2, loc='upper right')
plt.show()

In [ ]:
for i, j, k, l in zip(epochs, train_loss, val_loss, auc):
    print(f"Epoch: {i}, Train Loss: {j}, Val Loss: {k}, AUC: {l}")

Epoch: 1, Train Loss: 0.9755, Val Loss: 0.863, AUC: 0.5125
Epoch: 2, Train Loss: 0.8572, Val Loss: 0.8488, AUC: 0.5967
Epoch: 3, Train Loss: 0.8289, Val Loss: 0.8038, AUC: 0.6061
Epoch: 4, Train Loss: 0.8151, Val Loss: 0.7882, AUC: 0.6469
Epoch: 5, Train Loss: 0.8028, Val Loss: 0.7713, AUC: 0.6732
Epoch: 6, Train Loss: 0.7935, Val Loss: 0.7602, AUC: 0.7162
Epoch: 7, Train Loss: 0.7834, Val Loss: 0.7711, AUC: 0.7038
Epoch: 8, Train Loss: 0.781, Val Loss: 0.7495, AUC: 0.7261
Epoch: 9, Train Loss: 0.7736, Val Loss: 0.7489, AUC: 0.7117
Epoch: 10, Train Loss: 0.7671, Val Loss: 0.7518, AUC: 0.7289
Epoch: 11, Train Loss: 0.7637, Val Loss: 0.765, AUC: 0.7017
Epoch: 12, Train Loss: 0.7599, Val Loss: 0.7292, AUC: 0.7462
Epoch: 13, Train Loss: 0.759, Val Loss: 0.741, AUC: 0.7204
Epoch: 14, Train Loss: 0.7525, Val Loss: 0.7233, AUC: 0.7435
Epoch: 15, Train Loss: 0.7522, Val Loss: 0.7206, AUC: 0.754
Epoch: 16, Train Loss: 0.7465, Val Loss: 0.7252, AUC: 0.754
Epoch: 17, Train Loss: 0.7452, Val Loss:

Epoch: 34, Train Loss: 0.6434, Val Loss: 0.681, AUC: 0.7888
Epoch: 35, Train Loss: 0.6415, Val Loss: 0.6763, AUC: 0.7878
Epoch: 36, Train Loss: 0.6402, Val Loss: 0.6776, AUC: 0.7882
Epoch: 37, Train Loss: 0.6407, Val Loss: 0.6754, AUC: 0.7931
Epoch: 38, Train Loss: 0.6367, Val Loss: 0.6885, AUC: 0.7875
Epoch: 39, Train Loss: 0.6369, Val Loss: 0.692, AUC: 0.7831
Epoch: 40, Train Loss: 0.6338, Val Loss: 0.678, AUC: 0.7896
Epoch: 41, Train Loss: 0.6323, Val Loss: 0.6913, AUC: 0.7874
Epoch: 42, Train Loss: 0.6302, Val Loss: 0.6896, AUC: 0.7885
Epoch: 43, Train Loss: 0.6315, Val Loss: 0.6898, AUC: 0.7868
Epoch: 44, Train Loss: 0.6284, Val Loss: 0.6917, AUC: 0.7887
Epoch: 45, Train Loss: 0.6276, Val Loss: 0.6897, AUC: 0.7878
Epoch: 46, Train Loss: 0.627, Val Loss: 0.6872, AUC: 0.7892
Epoch: 47, Train Loss: 0.6247, Val Loss: 0.6951, AUC: 0.7875

In [ ]:
# for i, j, k, l in zip(epochs, train_loss, val_loss, auc):
#     print(f"Epoch: {i}, Train Loss: {j}, Val Loss: {k}, AUC: {l}")